loadgen/system_details.cc

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2019 The MLPerf Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+/// \file
+/// \brief Prints out system details.
+
+#include <array>
+#include <iostream>
+#include <memory>
+
+#include "logging.h"
+#include "utils.h"
+
+namespace mlperf {
+
+std::string RunCommand(const std::string& cmd) {
+  std::string result;
+  try {
+    std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd.c_str(), "r"), pclose);
```

Comment:
we want loadgen to work for all devices. pipe() might not be portable on mobile, tiny and windows

Reply:
@TheKanter @xyhuang any thought on this? What would be the best way to maintain portability?

Reply:
The mechanism should be there for most platforms, maybe ifdef is fine.